src/jdk.compiler/share/classes/com/sun/tools/javac/code/Types.java

```diff
@@ -1459,14 +1459,15 @@ public Boolean visitClassType(ClassType t, Type s) {
                 }
                 return t.tsym == s.tsym
                     && t.isReferenceProjection() == s.isReferenceProjection()
+                    && t.isValueProjection() == s.isValueProjection()
                     && visit(getEnclosingType(t), getEnclosingType(s))
```

Comment:
Likewise this should be checking that t.flavor == s.flavor. Will be followed up in JDK-8270882